cmake/Dependencies.cmake

```diff
@@ -1232,9 +1232,19 @@ if(USE_ROCM)
   # Currently only active for Ubuntu 20.04 and greater versions.
   if(UNIX AND EXISTS "/etc/os-release")
     file(STRINGS /etc/os-release OS_RELEASE)
-    string(REGEX REPLACE ".*NAME=\"([A-Za-z]+).*" "\\1" OS_DISTRO ${OS_RELEASE})
-    string(REGEX REPLACE ".*VERSION_ID=\"([0-9\.]+).*" "\\1" OS_VERSION ${OS_RELEASE})
-    if(OS_DISTRO STREQUAL "Ubuntu" AND OS_VERSION VERSION_GREATER_EQUAL "20.04")
+    set(DISTRO_NAME "")
```

Comment:
I dont think this change is required. 
Previously there was a bug and it is fixed by this commit 
https://github.com/ROCmSoftwarePlatform/pytorch/commit/0d1538dac18ce972da260f7ce21961c741866f76

With this change libtinfo will be properly picked in UB 22.04 and this change is currently in all active branches and

Reply:
I'm not so sure about the suitable choice here.
The current code actually is still broken when parsing the `/etc/os-release` file, and only works for Ubuntu.
The patched code is more complicated but with fewer flaws.

Ideally we should use `cmake_host_system_information` to get `DISTRIB_NAME`, but it requires cmake 3.22+

Reply:
I'm good with these changes.  IMHO the most important fix here was the regex matching from the beginning `"^NAME=` to avoid the PRETTY_NAME key later in the os-release file.

Reply:
FYI https://github.com/ROCmSoftwarePlatform/pytorch/commit/0d1538dac18ce972da260f7ce21961c741866f76 is the wrong fix.  Should have used `^` instead of `.*`